src/coins.cpp

```diff
@@ -51,10 +51,6 @@ CCoinsMap::iterator CCoinsViewCache::FetchCoin(const COutPoint &outpoint) const
         if (auto coin{base->GetCoin(outpoint)}) {
             ret->second.coin = std::move(*coin);
             cachedCoinsUsage += ret->second.coin.DynamicMemoryUsage();
-            if (ret->second.coin.IsSpent()) { // TODO GetCoin cannot return spent coins
```

Comment:
Now that we've removed it we should at least add an `Assume(!ret->second.coin.IsSpent());` here - or even an assert, since if we're not sure about this we shouldn't do it in the first place

Reply:
Done.